File: src/go/pe_0010.go
package main


// SCRIPT


func main() {
    // initialize vairables
    size := 2000000;
    half := size / 2
    prime := make([]bool, size)

    // set two as prime, begin sum
    prime[2] = true;
    gross := 2

    // all odds are potential primes
    for i := 3; i < size; i += 2 {
        prime[i] = true
    }

    // disregard multiples of primes
    i := 3
    for ; i < half; i += 2 {
        if prime[i] {
            gross += i
            for j := i * 2; j < size; j += i {
                prime[j] = false
            }
        }
    }

    // continue the sum
    for ; i < size; i += 2 {
        if prime[i] {
            gross += i
        }
    }

    // print out
    println(gross)
}

File: src/jl/pe_0010.jl
# SCRIPT


function main()
    # initialize vairables
    size = 2000000
    half = size ÷ 2
    prime = BitArray(undef, size)

    # set two as prime, begin sum
    prime[2] = true
    gross = 2
    
    # all odds are potential primes
    for i in range(3, size, step=2)
        prime[i] = true
    end
    
    # disregard multiples of primes
    i = 3
    while i < half
        if prime[i] == 1
            gross += i
            for j in range(i * 2, size, step=i)
                prime[j] = false
            end
        end
        i += 2
    end

    # continue the sum
    for i in range(i, size, step=2)
        if prime[i]
            gross += i
        end
    end

    # print out
    println(gross)
end

main()
